### ISSUE TITLE
API-23: MP Reactive Streams Operators 1.0 Epic

### ISSUE THREAD

**Issue by Emily-Jiang** [OpenLiberty/open-liberty#4089]
Supports MicroProfile Reactive specification in OL
https://github.com/eclipse/microprofile-reactive-streams/

Currently James Roper from Lightbend is doing an implementation in Lightbend github https://github.com/lightbend/microprofile-reactive-streams. Although the files missed the licenses. I asked him to put on Apache License v2 and he agreed.
For our implementation strategy, I would like to propose to directly pull in the zero dependency impl (https://github.com/lightbend/microprofile-reactive-streams/tree/master/zerodep) and get it integrated in OL.
This should be delivered an internal feature, to be used by the featured reactive-messaging

**Comment by marikaj123** [comment]
Not in 19200 beta.

**Comment by hutchig** [comment]
Yes, @marikaj123 - I am targetting 19300, and had assummed 19200 is done and dusted wrt to this process.

**Comment by hutchig** [comment]
AHA URL is at https://bigblue.aha.io/features/API-23

**Comment by hutchig** [comment]
Hi @gscottj  this feature has no user interface in any way. Is there a process by which I reaquest (and provide information for) and accessibility review so that you can add a focalApproved:accessibility tag please?

**Comment by gscottj** [comment]
Based on the description provided, it appears that this feature does not have any end-user user interfaces that require accessibility compliance testing. If this assumption is not correct, then let me know that required accessibility verification testing of the UI will be performed before it is shipped and what are the results from those tests.

**Comment by marikaj123** [comment]
According to @hutchig the feature has no requirements for translation and Globalization of any code artifacts/messages; therefore, I am approving it.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Please add the "FAT Complete" label once all code is delivered and the Feature Test Summary (#4087) has been completed. Thanks!

**Comment by hutchig** [comment]
@donbourne is there anything else I have to add to https://github.com/OpenLiberty/open-liberty/issues/4089#issuecomment-478625851 in order to request your focal point review?

**Comment by hutchig** [comment]
Dear @donbourne , is there anything I need to add tohttps://github.com/OpenLiberty/open-liberty/issues/4089#issuecomment-478625851 in order to initiate a servicability review of my feature?

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi @hutchig, thanks for completing the Feature Test Summary (#4087). I've finished my review and provided the test results from the SOE run are good on Monday I'll be able to grant the approval then.

**Comment by hutchig** [comment]
Dear @skasund,  I have placed the STE slides at https://ibm.box.com/s/lnkhhrh3kqouqtlvoctbzkw2rex9qrls
please could you provide STE review/approval as appropriate?

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Following up from my previous comment, your testing in the SOE is passing so I've added the FAT Focal Approval.

**Comment by skasund** [comment]
@hutchig  Thanks for the slides!  I've provided the STE approval.

**Comment by hutchig** [comment]
I have nothing more to add to https://github.com/openliberty/open-liberty/issues/6523 and have merged all my development PRs for this (including the kind=ga etc ones). 
I will close this epic once the kind=ga feature is in the master branch of the build. 

There are 3 more things related to this work not closed:
   associated article
  #6523  associated text in release 19003's blog
  #6674  the ship readiness checklist issue (which encompasses the two things above and a 
              "Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / PM)" - which I do not know how to close down?

I have raised https://github.ibm.com/dev-engage/blog-posts/issues/73
(and will create a linked PR) to cover the creation of the draft article
that discusses the feature.

**Comment by hutchig** [comment]
kind=ga is now in the master branch so I will close this epic.
If this is premature, feel free to re-open.
